# Google Play Store Exploratory Data Analysis (EDA)

This project analyzes the Google Play Store dataset to identify trends in app ratings, reviews, installs, and categories using Python.

In [303]:
# Pandas--Google Play Store App Analysis
import pandas as pd
import numpy as np
pd.set_option('Display.max_columns',None)

In [ ]:
df = pd.read_csv('googleplaystore.csv')
df

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up
...,...,...,...,...,...,...,...,...,...,...,...,...,...
10836,Sya9a Maroc - FR,FAMILY,4.5,38,53M,"5,000+",Free,0,Everyone,Education,"July 25, 2017",1.48,4.1 and up
10837,Fr. Mike Schmitz Audio Teachings,FAMILY,5.0,4,3.6M,100+,Free,0,Everyone,Education,"July 6, 2018",1.0,4.1 and up
10838,Parkinson Exercices FR,MEDICAL,NaN,3,9.5M,"1,000+",Free,0,Everyone,Medical,"January 20, 2017",1.0,2.2 and up
10839,The SCP Foundation DB fr nn5n,BOOKS_AND_REFERENCE,4.5,114,Varies with device,"1,000+",Free,0,Mature 17+,Books & Reference,"January 19, 2015",Varies with device,Varies with device


## Find the total number of apps in the google playstore

In [305]:
# Task1: Find the total number of apps in the google playstore
# To find the total number of apps we use shape attribute
print('Total number of apps in playstore =',df.shape[0]) # The 0 refers to the rows in the csv file

Total number of apps in playstore = 10841


## Find the total number of columns in each app of google playstore

In [306]:
# Task2: Find the total number of columns in each app of google playstore
# To find the total number of columns for each app we use shape attribute
print('Total number of columns in each app of google playstore =',df.shape[1]) # The 1 refers to total number of columns in csv file

Total number of columns in each app of google playstore = 13


## Display top 5 rows of the dataset

In [307]:
# Task3: Display top 5 rows of the dataset
# Using head() function we can access top 5 rows of a dataset
# We can also use loc or iloc functions 
print('-----------------------------------------------------------------------------------------')
print('Top 5 Rows of the Dataset')
print('-----------------------------------------------------------------------------------------')
print(df.head())
print('-----------------------------------------------------------------------------------------')

-----------------------------------------------------------------------------------------
Top 5 Rows of the Dataset
-----------------------------------------------------------------------------------------
                                                 App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M   

## Check the last 3 rows of the Dataset

In [308]:
# Task 4: Check the last 3 rows of the Dataset
# To access the last 3 rows of dataset tail() function can be used
print('-----------------------------------------------------------------------------------------')
print('Least 3 Rows of the Dataset')
print('-----------------------------------------------------------------------------------------')
print(df.tail(3))
print('-----------------------------------------------------------------------------------------')


-----------------------------------------------------------------------------------------
Least 3 Rows of the Dataset
-----------------------------------------------------------------------------------------
                                                 App             Category  \
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference 

## Find the shape of our Dataset

In [309]:
# Task 5: Find the shape of our Dataset (Number of Rows and Number of Columns)
# We use shape attribute to access number of columns and rows in a dataset

print('Number of Rows and Columns =',df.shape)

Number of Rows and Columns = (10841, 13)


## Get information about our dataset like total number of rows and total number of columns and Datatypes of Each Column and Memory Requirement

In [310]:
# Task 6: Get information about our dataset like total number of rows and total number of columns Datatypes of Each Column and Memory Requirement
# To access the information like number of non null values dtype of each column and shape of the data set
# We use info() function
print('--------------------------------------------------')
print('Information about the Dataset')
print('--------------------------------------------------')
df.info()
print('--------------------------------------------------')

--------------------------------------------------
Information about the Dataset
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB
-------------------------------

## Get overall statistics of Dataset 

In [311]:
'''
# Task 7: Get overall statistics of Dataset
           OR 
# Get the descriptive statistics about Google Play Store
'''
# describe() is a function used to deisplay the numerical stats of dataset
# df.describe(include = 'all') used to display over all stats
print('Statistical description of Google Playstore')

df.describe(include = 'all')

Statistical description of Google Playstore


,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
count,10841,10841,9367.000000,10841,10841,10841,10840,10841,10840,10841,10841,10833,10838
unique,9660,34,NaN,6002,462,22,3,93,6,120,1378,2832,33
top,ROBLOX,FAMILY,NaN,0,Varies with device,"1,000,000+",Free,0,Everyone,Tools,"August 3, 2018",Varies with device,4.1 and up
freq,9,1972,NaN,596,1695,1579,10039,10040,8714,842,326,1459,2451
mean,NaN,NaN,4.193338,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
std,NaN,NaN,0.537431,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
min,NaN,NaN,1.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25%,NaN,NaN,4.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
50%,NaN,NaN,4.300000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
75%,NaN,NaN,4.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


## Obtain total number of apps contain astrology

In [312]:
# Task 8: Obtain total number of apps contain astrology
# To obtain number of apps containing Astrology
# Access the apps containing Astrology-->Return an ndarray
# len of the array will give the total number of apps
print('Apps containing Astrology')
print('---------------------------------------------------------')
print((df[(df['App'].str.contains('Astrology'))]['App']).values)
print('---------------------------------------------------------')
print('Number of apps containing Astrology =',len((df[(df['App'].str.contains('Astrology'))]['App']).values))

Apps containing Astrology
---------------------------------------------------------
['Horoscopes – Daily Zodiac Horoscope and Astrology'
 'သိင်္ Astrology - Min Thein Kha BayDin'
 'iHoroscope - 2018 Daily Horoscope & Astrology']
---------------------------------------------------------
Number of apps containing Astrology = 3


## Find average app Rating in Google Play Store

In [313]:
# Task 9: Find the average app rating in google play store
# First access the Rating column --> Using mean() helps to obtain average rtating of the apps
print('Average App rating =',df['Rating'].mean())

Average App rating = 4.193338315362443


## Find the total number of unique category apps

In [314]:
# Task 10: Find the number of unique values Category
# Method 1:
# Access the Category column then using unique() returns ndarray of unique values --> then len(ndarray) gives number of unique values
print('Number of unique values in Category =',len(df['Category'].unique()))

Number of unique values in Category = 34


In [315]:
# Method 2:
# Access the Category column and use nunique
print('Number of unique values in Category =',df['Category'].nunique())

Number of unique values in Category = 34


## Find the Category with the Highest Average Rating

In [316]:
# Task 11 : Find the category with the highest average rating
# Group by category and calculate mean ratings sort the values in descending order access the top Category
# This give a series dtype with Category and top rating
# Accessing the index of that series gives us Category name 
rating = df.groupby('Category')['Rating'].mean().sort_values(ascending=False).head(1)
print('Highest average rating of app')
print('---------------------------------------------------------------------------------')
print(rating)
print('---------------------------------------------------------------------------------')
print('Category with highest average rating')
print('---------------------------------------------------------------------------------')
rating.index[0]

Highest average rating of app
---------------------------------------------------------------------------------
Category
1.9    19.0
Name: Rating, dtype: float64
---------------------------------------------------------------------------------
Category with highest average rating
---------------------------------------------------------------------------------


'1.9'

## Find the number of apps with 5 star rating

In [317]:
# Task 12: Find the number of apps with 5 star rating
# We filter the dataset using condition
# Access the Apps with Rating of 5 retuns a data set --> len() the data gives number of apps
# sum() dataset can be used to find number of apps rated with 5 stars
apps = df[df['Rating']==5]
print('Number of apps achieved 5 star rating in google playstore =',len(apps))
print('------------------------------------------------------------------------')
rating = df['Rating']==5 # Returns bool
print('Number of apps achieved 5 star rating in google playstore =',(rating).sum())

Number of apps achieved 5 star rating in google playstore = 274
------------------------------------------------------------------------
Number of apps achieved 5 star rating in google playstore = 274


## Data Cleaning and Preprocessing

The raw dataset contained inconsistent formats and invalid entries. The following corrections were performed to prepare the data for analysis.

## Find the average value of reviews

In [320]:
# Task 13: Find the average value of reviews
# To obtain average reviews of apps in google playstore we should access the Reviews column
# Replace the strings with numerical values
# Convert the whole Reviews data to Float then find the mean
df['Reviews'].replace('3.0M','3000000',inplace=True)
print('Average value of Reviews =',df['Reviews'].astype(float).mean())

Average value of Reviews = 444388.6535374965


## Find the total number of Free paid apps

In [321]:
# Task 14: Find the total number of free paid apps
# To find the number of free apps available in the google playstore
# Access the Type(it describes the app paid/free) column
# Search for free then -->len(free_apps)
free_apps = df[df['Type']=='Free']
print('Number of apps available for free in google playstore =',len(free_apps))

Number of apps available for free in google playstore = 10039


In [322]:
print('Total number of paid apps available in google playstore',(df['Type']=='Paid').sum())

Total number of paid apps available in google playstore 800


In [323]:
df['Type'].value_counts()

Type
Free    10039
Paid      800
0           1
Name: count, dtype: int64

## Find the app with maximum reviews

In [324]:
# Task 15: Find the app with maximum reviews
# Access the Reviews column and then find the max values in reviews column
# Using the max value we can access the App name
# Since it is a series dtype we use values attribute to obtain name
highest = df['Reviews'].max()
print('Maximum reviews =',highest)
print('--------------------------------------------------')
print('App with maximum reviews =', df[df['Reviews']==df['Reviews'].max()]['App'])

Maximum reviews = 9992
--------------------------------------------------
App with maximum reviews = 2989    GollerCepte Live Score
Name: App, dtype: object


## Display top 5 apps having highest reviews

In [325]:
# Task 16: Find the top 5 apps having highest reviews
# To obtain the top 5 apps with highest reviews should sort reviews column values in descending order
# Obtain top 5 apps index values using head() function
# Using loc or iloc --> df.iloc[index] gives us apps data
# Then access the app column

ind = df['Reviews'].astype(float).sort_values(ascending=False).head().index
print('Top 5 Apps with highest reviews')
print('-------------------------------------------------------')
print(df.iloc[ind]['App'])

Top 5 Apps with highest reviews
-------------------------------------------------------
2544              Facebook
3943              Facebook
381     WhatsApp Messenger
336     WhatsApp Messenger
3904    WhatsApp Messenger
Name: App, dtype: object


## Display top 5 apps having maximum number of installs

In [326]:
# Task 17: Find the top 5 apps with highest number of installs
# Obtain the installs column from dataset
# Convert the Installs column values to int 
# Replace ',' and '+' using replace() function
# Sort the values in installs column, using head() obtain top 5 installs and using index attribute obtain indices of the apps
# Using those indices then obtain the top 5 apps with maximum installs
df['Installs'] = df['Installs'].str.replace(',','')
df['Installs'] = df['Installs'].str.replace('+','')
df['Installs'] = df['Installs'].str.replace('Free','0')
df['Installs'] = df['Installs'].astype(int)


In [327]:
index_val = df['Installs'].sort_values(ascending = False).head().index

In [328]:
df.loc[index_val]['App']

5856    Google Play Games
5395        Google Photos
2853        Google Photos
2884        Google Photos
4170         Google Drive
Name: App, dtype: object

## Find the average rating of free and paid apps

In [329]:
# Task 18: Find the average rating of free and paid apps
# To obtain the average Rating of free and paid apps we use groupby() finction
# groupby('Type') and obtain the mean for Rating column
print('Average rating of Free and paid Apps')
print('-----------------------------------------')
print(df.groupby('Type')['Rating'].mean())

Average rating of Free and paid Apps
-----------------------------------------
Type
0       19.000000
Free     4.186203
Paid     4.266615
Name: Rating, dtype: float64


## Conclusion

- Family and Game categories contained the most applications.
- Most apps had ratings between 4.0 and 4.5.
- Apps with more installs generally received more reviews.
- The majority of apps on the Play Store were free.